typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
             TypeScript rules outside of the recommended configuration
         */
-        // '@typescript-eslint/naming-convention': 'error',
-        // '@typescript-eslint/no-base-to-string': 'error',
-        // '@typescript-eslint/no-confusing-void-expression': 'error',
-        // '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error',
-        // '@typescript-eslint/no-unnecessary-condition': 'error',
-        // '@typescript-eslint/no-unnecessary-qualifier': 'error',
-        // '@typescript-eslint/no-unnecessary-type-arguments': 'error',
-        // '@typescript-eslint/non-nullable-type-assertion-style': 'error',
-        // '@typescript-eslint/prefer-includes': 'error',
-        // '@typescript-eslint/prefer-nullish-coalescing': 'error',
-        // '@typescript-eslint/prefer-readonly': 'error',
-        // '@typescript-eslint/prefer-readonly-parameter-types': 'error',
-        // '@typescript-eslint/prefer-reduce-type-parameter': 'error',
-        // '@typescript-eslint/prefer-string-starts-ends-with': 'error',
-        // '@typescript-eslint/promise-function-async': 'error',
-        // '@typescript-eslint/require-array-sort-compare': 'error',
-        // '@typescript-eslint/strict-boolean-expressions': 'error',
-        // '@typescript-eslint/switch-exhaustiveness-check': 'error'
+        '@typescript-eslint/naming-convention': 'error',
```

Comment:
This rule is erroring on Enum's _not_ being camel case, which is against the norm that I've seen in our TS codebases. 
>`Enum Member name 'Create' must match one of the following formats: camelCase` 

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md

There's a very extensive config for this rule that we probably want to go over in a meeting.

Reply:
At a minimum we should change enums to be `PascalCase` and not `YELLING_CASE` or `snake_case`, or `camelCase`. @mure will audit the remainder of the config options this week. If he doesn't find time, we can consider disabling this rule temporarily and revisiting this in another PR

Reply:
**Proposal:**
To match what seems to be a standard convention, we should force enums to be PascalCase rather than camelCase.
Building on the default config:
```
{
  "@typescript-eslint/naming-convention": [
    "error",
  {
    selector: 'default',
    format: ['camelCase'],
    leadingUnderscore: 'allow',
    trailingUnderscore: 'allow',
  },
  {
    selector: 'variable',
    format: ['camelCase', 'UPPER_CASE'],
    leadingUnderscore: 'allow',
    trailingUnderscore: 'allow',
  },
  {
    selector: 'enumMember',
    format: ['PascalCase']
  },
  {
    selector: 'typeLike',
    format: ['PascalCase'],
  },
]}
```

Interestingly, there's an issue that points out this inconsistency, but it was closed (incorrectly I think): https://github.com/typescript-eslint/typescript-eslint/issues/3260

Reply:
@mure will update the comment to emphasize what changes from the default. We may consider re-opening / posting a new issue.

Reply:
It's also worth investigating whether we can only override the areas that are changing rather than copying the base configuration.

We agree on the direction, @mure just needs to update the config with comments or less duplicate information.

Reply:
**Updated proposal**
```js
/* 
    The default configuration for this rule requires that enum members
    follow the camelCase style, which doesn't match the existing convention
    within NI and the broader TS community to use PascalCase. The rest of the
    configuration is identitical to the default.
*/ 
"@typescript-eslint/naming-convention": [
    "error",
    {
        selector: "enumMember",
        format: ["PascalCase"],
    },
    {
        selector: "default",
        format: ["camelCase"],
        leadingUnderscore: "allow",
        trailingUnderscore: "allow",
    },
    {
        selector: "variable",
        format: ["camelCase", "UPPER_CASE"],
        leadingUnderscore: "allow",
        trailingUnderscore: "allow",
    },
    {
        selector: "typeLike",
        format: ["PascalCase"],
    },
],
```

It doesn't appear that it's possible to extend the default configuration, so we will have to duplicate it.

Reply:
@rajsite this thread is ready to close. @LukasKall hasn't voted on the latest proposal but it's aligned with what we've talked about before so it's likely he's ok with it. If not, we can revisit.